### ISSUE TITLE
FAT kv_ro example failure (fat-filesystem compilation error)

### ISSUE THREAD

**Issue by lambdaxdotx** [mirage/mirage-skeleton#73]
Trying the command
`$ env FS=fat mirage configure`
I get the following compilation error from fat-filesystem:
```
# opam-version 1.2.0
# os           linux
# command      make
# compiler     system (4.01.0)
# exit-code    2
### stdout ###
# ...[truncated]
# Error: Signature mismatch:
#        ...
#        Type declarations do not match:
#          type t = Io_page.t
#        is not included in
#          type t =
#              (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout)
#              Bigarray.Array1.t
# Command exited with code 2.
# Makefile:19: recipe for target 'build' failed
### stderr ###
# W: Cannot find source file matching module 'fat' in library fat
# E: Failure("Command ''/usr/bin/ocamlbuild' lib/fat.cma lib/fat.cmxa lib/fat.a lib/fat.cmxs fat/main.native shell/main.native lib_test/test.native -tag debug -j 4' terminated with error code 10")
```
Any idea?

**Comment by dsheets** [comment]
This is due to a change in io-page since 1.3.0. You could downgrade to io-page before 1.3.0 or modify ocaml-fat probably here <https://github.com/mirage/ocaml-fat/blob/9d7abc383ebd9874c2d909331e2fb3cc08d7304b/fat/impl.ml#L37> and here <https://github.com/mirage/ocaml-fat/blob/9d7abc383ebd9874c2d909331e2fb3cc08d7304b/lib_test/test.ml#L24> with code similar to this <https://github.com/mirage/ocaml-dns/commit/6f35270e5962e21d75bf9e133810b9ca1a3530b2>.

**Comment by dsheets** [comment]
On further investigation, that's not sufficient to fix. Investigating.

**Comment by dsheets** [comment]
Two potential fixes have been submitted as PRs. As the `Io_page` interface is in a bit of flux recently, these may not be the preferred solutions but either should get you on your way again. Also fixed by talex5/ocaml-fat@d393c629397fbaa5a81d26b0d1e953de19237525.

**Comment by dsheets** [comment]
This has now been fixed in opam by ocaml/opam-repository#3493 and ocaml/opam-repository#3494. Please `opam update` and re-open this issue if it still does not work.

Thanks very much for your report! :-)
